flink-ml-lib/src/main/java/org/apache/flink/ml/classification/linear/LogisticGradient.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.classification.linear;
+
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.api.java.tuple.Tuple3;
+import org.apache.flink.ml.common.linalg.BLAS;
+
+import java.io.Serializable;
+
+/** Logistic gradient. */
+public class LogisticGradient implements Serializable {
+    private static final long serialVersionUID = 1178693053439209380L;
+
+    /** L1 regularization term. */
+    private final double l1;
+
+    /** L2 regularization term. */
+    private final double l2;
+
+    public LogisticGradient(double l1, double l2) {
+        this.l1 = l1;
+        this.l2 = l2;
+    }
+
+    /**
+     * Computes loss and weightSum on a set of samples.
+     *
+     * @param labeledData a sample set of train data.
+     * @param coefficient model parameters.
+     * @return loss and weightSum.
+     */
+    public final Tuple2<Double, Double> computeLoss(
+            Iterable<Tuple3<Double, Double, double[]>> labeledData, double[] coefficient) {
+        double weightSum = 0.0;
+        double lossSum = 0.0;
+        double loss;
+        for (Tuple3<Double, Double, double[]> dataPoint : labeledData) {
+            loss = computeLoss(dataPoint, coefficient);
+            lossSum += loss * dataPoint.f0;
+            weightSum += dataPoint.f0;
+        }
+        if (Double.compare(0, Math.abs(l1)) != 0) {
```

Comment:
It looks like the LogisticRegression added in this PR asks users to separately set l1 and l2 parameter. Both values can be non-zero. And the loss is the sum of both loss.

It looks like most (if not all) logistic regression algorithm (e.g. [1] and [2]) uses one type of loss function (either l1 or l2), or use one parameter with value in range [0, 1] to specify the loss function (e.g. Spark's approach). So the approach used here seems to be different from common practice of how loss function of logistic regression is specified.

Could we follow the same approach as Spark, since it seems to be more consistent with commonly used logistic regression algorithm? 

[1] https://en.wikipedia.org/wiki/Logistic_regression
[2] https://www.section.io/engineering-education/understanding-loss-functions-in-machine-learning/

Reply:
Thanks for the comments. L1 is removed.